src/homepageExperience/containers/ArduinoWizard.tsx

```diff
@@ -64,46 +64,38 @@ export class ArduinoWizard extends PureComponent<{}, State> {
   }
 
   handleNextClick = () => {
-    this.setState(
+    this.setState({
+      currentStep: Math.min(
+        this.state.currentStep + 1,
+        HOMEPAGE_NAVIGATION_STEPS_ARDUINO.length
+      ),
+    })
+    event(
+      'firstMile.arduinoWizard.next.clicked',
+      {},
       {
-        currentStep: Math.min(
-          this.state.currentStep + 1,
-          HOMEPAGE_NAVIGATION_STEPS_ARDUINO.length
+        clickedButtonAtStep: normalizeEventName(
+          HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep - 1].name
+        ),
+        currentStep: normalizeEventName(
+          HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep].name
         ),
-      },
-      () => {
-        event(
-          'firstMile.arduinoWizard.next.clicked',
-          {},
-          {
-            clickedButtonAtStep: normalizeEventName(
-              HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep - 1].name
-            ),
```

Comment:
Why do we take it out of the setState callback?

Reply:
We take it out because the event is expecting the previous state rather than the updated state.

Reply:
Then i think we should put that before the setState, to make that explicitly clear.

Reply:
I agree, it's probably better to put it before the call to `setState` so we don't have to do wonky index math when firing the event

Reply:
I decided instead to not use indexing at all to avoid messiness. See the new commit where we only report the current step rather than the index. This is what we used for previous wizards so could be better for consistency.

Reply:
Hmm.. i am a little concerned about replacing the normalizeEventName function. It seems like a good addition for consistency.

Plus, we're going from step names to step numbers, which makes it more confusing to people who are going to read the events. like, 

```
clickedButtonAtStep: Finish
clickedButtonAtStep: 9
```

the first event name is better to follow.

Reply:
Or am I misunderstanding the change?